src/Test/Utilities/Portable/CommonTestBase.cs

```diff
@@ -526,5 +528,268 @@ internal static SignatureDescription Signature(string fullyQualifiedTypeName, st
+                // and we can't gurantee it will return same operation tree
+                if (child == null ||
+                    child.Kind == OperationKind.InvalidExpression ||
+                    child.Kind == OperationKind.InvalidStatement)
+                {
+                    continue;
+                }
+
+                yield return child;
+
+                foreach (var nested in GetOperationsUptoInvalidOperation(child))
+                {
+                    yield return nested;
+                }
+            }
+        }
+
+        private static void VerifyOperationTreeSpine(
+            SemanticModel semanticModel, HashSet<IOperation> set, SyntaxNode node, HashSet<string> output)
+        {
+            while (node != semanticModel.Root)
+            {
+                if (!IsIgnoredNode(node))
+                {
+                    var operation = semanticModel.GetOperationInternal(node);
+                    Assert.NotNull(operation);
+
+                    // all operation from same sub tree must belong to same operation tree
+                    // except OperationKind.None and OperationKind.InvalidExpression and InvalidStatement
+                    // for those kinds, we can't guarantee it will share same tree
+                    if (operation != null &&
+                        operation.Kind != OperationKind.None &&
+                        operation.Kind != OperationKind.InvalidExpression &&
+                        operation.Kind != OperationKind.InvalidStatement &&
+                        operation.Kind != OperationKind.PlaceholderExpression)
+                    {
+                        Assert.True(set.Contains(operation));
+                    }
+                }
+
+                node = node.Parent;
+            }
+        }
+
+        private static bool IsIgnoredNode(SyntaxNode node)
+        {
+            // this should be removed once this is fixed
+            // https://github.com/dotnet/roslyn/issues/21187
+            // basically, for these node. GetOpeartion will return null 
+            // even though GetOperation returns IOperation for its child (syntax node)
+            // violating our assumption that all spine once a node has IOperation should
+            // have an IOperation
+            if (node is CSharp.CSharpSyntaxNode csNode)
+            {
+                switch (csNode.Kind())
+                {
+                    case CSharp.SyntaxKind.VariableDeclarator:
+                        return csNode.Parent?.Kind() == CSharp.SyntaxKind.VariableDeclaration;
+                    case CSharp.SyntaxKind.EqualsValueClause:
+                        return csNode.Parent?.Kind() == CSharp.SyntaxKind.VariableDeclarator &&
+                               csNode.Parent?.Parent?.Kind() == CSharp.SyntaxKind.VariableDeclaration;
+                    case CSharp.SyntaxKind.IdentifierName when csNode.ToString() == "E":
```

Comment:
https://github.com/dotnet/roslyn/pull/20960 should fix IEventAssignment and IEventReference issue.